meson.build

```diff
@@ -0,0 +1,169 @@
+project('xviewer', 'c', version: '3.2.1', meson_version: '>= 0.49')
+
+version = meson.project_version()
+
+xviewer_conf = configuration_data()
+xviewer_conf.set_quoted('VERSION', version)
+xviewer_conf.set_quoted('GETTEXT_PACKAGE', meson.project_name())
+xviewer_conf.set('HAVE_INTROSPECTION', 1)
+
+# directories
+prefix = get_option('prefix')
+datadir = prefix / get_option('datadir')
+libdir = prefix / get_option('libdir')
+localedir = prefix /  get_option('localedir')
+
+xviewer_pkgdatadir = datadir / meson.project_name()
+xviewer_pkglibdir = libdir / meson.project_name()
+# workaround for plugins: keep old lib dir without arch-triplet
+xviewer_pluginsdir = prefix / 'lib' /  meson.project_name() / 'plugins'
+
+xviewer_conf.set_quoted('LIBDIR', libdir)
+xviewer_conf.set_quoted('XVIEWER_DATA_DIR', xviewer_pkgdatadir)
+xviewer_conf.set_quoted('XVIEWER_LOCALE_DIR', localedir)
+xviewer_conf.set_quoted('XVIEWER_PLUGIN_DIR', xviewer_pluginsdir)
+
+src_root = meson.source_root()
+po_dir = join_paths(src_root, 'po')
+
+# options
+enable_docs = get_option('docs')
+enable_exempi = get_option('enable_exempi')
+enable_exif = get_option('enable_exif')
+enable_jpeg = get_option('enable_jpeg')
+enable_lcms = get_option('enable_lcms')
+enable_rsvg = get_option('enable_rsvg')
+
+gnome = import('gnome')
+i18n = import('i18n')
+pkgconfig = import('pkgconfig')
+
+# dependencies
+glib = dependency('glib-2.0', version: '>= 2.38.0')
+gtk = dependency('gtk+-3.0')
+libpeas = dependency('libpeas-1.0', version: '>= 0.7.4')
+libpeas_gtk = dependency('libpeas-gtk-1.0', version: '>= 0.7.4')
+pixbuf = dependency('gdk-pixbuf-2.0', version: '>= 2.19.1')
+X11 = dependency('x11')
+zlib = dependency('zlib')
+cinnamon_desktop = dependency('cinnamon-desktop', version: '>= 3.2.0')
+xapp = dependency('xapp', version: '>= 1.9.0')
+
+intltool_merge = find_program('intltool-merge')
+cc = meson.get_compiler('c')
+
+# on some systems we need to find the math lib to make sure it builds
+math = cc.find_library('m')
```

Comment:
On some systems, meaning it should be `required: false` so that systems which don't have it, still work.

Any system will either have it, or not need it -- it is part of the base system and distributed with libc.

https://mesonbuild.com/howtox.html#add-math-library-lm-portably

Reply:
Thanks for the explanation, this makes sense to me now :) Fixed.